ui/src/app/components/common/datagrid/filters/boolean-filter/boolean-filter.component.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AfterViewInit, Component, Input, OnDestroy } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { ClrDatagridFilterInterface } from '@clr/angular';
+import { EqualsMultipleFilterAttributes } from '../../../../../models/search/equalsMultipleFilterAttributes.model';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-boolean-filter',
+  templateUrl: './boolean-filter.component.html',
+  styleUrls: ['./boolean-filter.component.scss'],
+})
+export class BooleanFilterComponent implements ClrDatagridFilterInterface<any>, AfterViewInit, OnDestroy {
+  @Input() removeFiltersSubject: Subject<any>;
+  @Input() property: string;
+  @Input() value;
+  isTrueSelected = false;
+  isFalseSelected = false;
+
+  changes = new Subject<any>();
+
+  modelChanges: Subject<any> = new Subject<any>();
+  modelSubscription: Subscription;
```

Comment:
I know we have this in the `MultipleStatusFilter` as well. But I think we don't need the indirection with the `modelChanges` subject here (and also not in the `MultipleStatusFilter`). IIRC, we only need the `modelChanges` subject for the `debounceTime` to add some delay for the filters which involve users typing. But here, we only have checkboxes, so it doesn't make sense to delay the request.

So I think, you can call `changes.next()` directly instead of `modelChanges.next()`

@jozefbakus What do you think?

Reply:
fixed thanks 👍🏽 .